pkgs/build-support/docker/default.nix

```diff
@@ -952,15 +952,24 @@ rec {
           # when the number of layers equals:
           #      maxLayers-1, maxLayers, and maxLayers+1, 0
           store_layers="$(
-            paths |
-              jq -sR '
-                rtrimstr("\n") | split("\n")
-                  | (.[:$maxLayers-1] | map([.])) + [ .[$maxLayers-1:] ]
+            if [[ $availableLayers -eq 1 ]]; then
+              paths |
+                jq -sR '
+                  rtrimstr("\n") | split("\n")
+                  | (.[:0] | map([.])) + [ .[0:] ]
                   | map(select(length > 0))
-              ' \
-                --argjson maxLayers "$availableLayers"
+                '
+            else
+              paths |
+                jq -sR --argjson maxLayers $availableLayers '
+                  rtrimstr("\n") | split("\n")
+                  | (.[:$maxLayers-2] | map([.])) + [ .[$maxLayers-2:-1] ] + [ [.[-1]] ]
+                  | map(select(length > 0))
+                '
+            fi
```

Comment:
Please add a comment documenting what's happening.

Your PR description is great, but it's always nice to have rationale live next to the code as well :)

Reply:
Agreed - the comment needed updating with this new method. Added!